core/src/server/rpc_module.rs

```diff
@@ -786,9 +746,71 @@ impl SubscriptionSink {
 		self.inner_send(msg).map_err(Into::into)
 	}
 
+	/// Consumes the `SubscriptionSink` and reads data from the `stream` and sends back data on the subscription
+	/// when items gets produced by the stream.
+	///
+	/// Returns `Ok(())` if the stream or connection was terminated.
+	/// Returns `Err(_)` if one of the items couldn't be serialized.
+	///
+	/// # Examples
+	///
+	/// ```no_run
+	///
+	/// use jsonrpsee_core::server::rpc_module::RpcModule;
+	///
+	/// let mut m = RpcModule::new(());
+	/// m.register_subscription("sub", "_", "unsub", |params, mut sink, _| {
+	///     let stream = futures_util::stream::iter(vec![1_u32, 2, 3]);
+	///     tokio::spawn(sink.consume_and_streamify(stream));
+	///     Ok(())
+	/// });
+	/// ```
+	pub async fn consume_and_streamify<S, T>(mut self, mut stream: S) -> Result<(), Error>
+	where
+		S: Stream<Item = T> + Unpin,
+		T: Serialize,
+	{
+		let mut close_stream = self.close.clone();
+		let mut item = stream.next();
+		let mut close = close_stream.next();
+
+		loop {
+			match futures_util::future::select(item, close).await {
+				Either::Left((Some(i), c)) => {
+					match self.send(&i) {
+						Ok(_) => (),
+						Err(Error::SubscriptionClosed(close_reason)) => {
+							self.close(&close_reason);
+							break Ok(());
+						}
+						Err(err) => {
+							tracing::warn!("Subscription got error: {:?} terminating task", err);
+							break Err(err);
```

Comment:
this is most likely a bug when serializing T fails, should we have a flag for this when the exit, just log an continue, or something else?

The error is not really possible to handle but if you require each element to be processed in order, ignore might break expectations so I just terminated the "task" when an error is detected.

Reply:
I think you made the right choice. We could also panic I guess but let's terminate for now.